File: codechef/MAR14_BINTOUR.cpp
/**********************************************************************/
//  																   /	
//	INFINITUM 					                                       /
//	Rakesh Mahadasa													   /
//	National Institute of Technology , Calicut						   /
//																	   /
/**********************************************************************/


//header files

#include<iostream>
#include<string.h>
#include<algorithm>
#include<vector>
#include<stack>
#include<queue>
#include<cstdio>
#include<iomanip>
#include<cmath>

//typedefs

typedef unsigned long long int ull;
typedef long long int ll;
typedef long int li;

using namespace std;

ull fact(ll a){
ull res = 1;
for(int i = 1; i <= a; i ++){
res = (res*i)%1000000009;
}
return res;
}

int main(){
int k;
cin>>k;
ull n = pow(2,k);
for(li i = 1; i < n/2 ; i++) cout<<0<<endl;
ull fct = fact(n/2);
fct = (((fct*fct)%1000000009)*2)%1000000009;
ull den = 1;
cout<<fct<<endl;
for(li i = n/2+1; i <= n ; i++){
if((fct*(i-1)) % (i-(n/2)) != 0) {cout<<"fuck"; cout<<i<<endl; return 0;}
fct = ((fct*(i-1))/(i-(n/2)))%1000000009;
if(i==n)
cout<<(fct)%1000000009<<endl;

}
return 0;
}

File: Codechef/MAR14_BINTOUR.py
def fact(a):
	res = 1
	for x in range(1,a+1):
		res = (res*x)%1000000009
	return res	
k=input()
n=pow(2,k)
fct = fact(n/2)
fct = (fct*fct*2)%1000000009
for x in range(1,n/2):
	print(0)
den = n/2
for x in range(n/2,n+1):
		if x == n/2:
			print(fct)
		else:
			print((fct*den)%1000000009)
			den = (((den*x)%1000000009)*pow((x+1-(n/2)),1000000007,1000000009))%1000000009
